scripts/pythondistdeps.py

```diff
@@ -350,25 +369,34 @@ def normalize_name(name):
                     deps.insert(0, Requirement.parse('setuptools'))
                 # add requires/recommends based on egg/dist metadata
                 for dep in deps:
-                    if normalized_names_require_pep503:
-                        dep_normalized_name = normalize_name(dep.project_name)
-                    else:
-                        dep_normalized_name = dep.key
-
-                    if args.legacy:
-                        name = 'pythonegg({})({})'.format(pyver_major, dep.key)
-                    else:
-                        if args.majorver_only:
-                            name = 'python{}dist({})'.format(pyver_major, dep_normalized_name)
+                    # Even if we're requiring `foo[bar]`, also require `foo`
+                    # to be safe, and to make it discoverable through
+                    # `repoquery --whatrequires`
+                    extras_suffixes = [""]
+                    if args.require_extras_subpackages and dep.extras:
+                        # A dependency can have more than one extras,
+                        # i.e. foo[bar,baz], so let's go through all of them
```

Comment:
```suggestion
                        # i.e. foo[bar,baz], so let's go through all of them
                        # we also always keep the non-extra in, so packages require foo when they require foo[bar]
                        # that is consistent with pip behavior and friendlier for repoqueries
```

Reply:
This is covered by the comment a few lines above. Do you think it's better to move it here?

Reply:
Ie missed it. Ignore this.